### PR TITLE
Fix bugs with einsums

### PR DESCRIPTION
**Pull request by cgbriggs99**
Adding some tests and fixing the bugs I find.

### CODE REVIEW

File: tests/QChemTests.cpp [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+    H_sym[3]       = symm_temp2(Range{5, 7}, Range{5, 7});
+
+    REQUIRE_NOTHROW(einsum(Indices{index::i, index::j, index::k, index::l}, &TEI_temp1, Indices{index::m, index::j, index::k, index::l},
+                           TEI, Indices{index::m, index::i}, Symm));
+    REQUIRE_NOTHROW(einsum(Indices{index::i, index::j, index::k, index::l}, &TEI_temp2, Indices{index::i, index::m, index::k, index::l},
+                           TEI_temp1, Indices{index::m, index::j}, Symm));
+    REQUIRE_NOTHROW(einsum(Indices{index::i, index::j, index::k, index::l}, &TEI_temp1, Indices{index::i, index::j, index::m, index::l},
+                           TEI_temp2, Indices{index::m, index::k}, Symm));
+    REQUIRE_NOTHROW(einsum(Indices{index::i, index::j, index::k, index::l}, &TEI_temp2, Indices{index::i, index::j, index::k, index::m},
+                           TEI_temp1, Indices{index::m, index::l}, Symm));
```
Comment (cgbriggs99) [inline tests/QChemTests.cpp:240]: Is this how to adapt the two electron integrals to point group symmetry?

**Comment by jturney** [comment]
Does C = 2.0 A - B also work? The mix and matching.

**Comment by cgbriggs99** [comment]
It should, I just didn't think to add a test.

**Comment by cgbriggs99** [comment]
Fixes #81

**Pull request merged**